robolectric/src/test/java/org/robolectric/shadows/ShadowAccessibilityManagerTest.java

```diff
@@ -15,49 +16,55 @@
 
   @Before
   public void setUp() throws Exception {
-    accessibilityManager = (AccessibilityManager) RuntimeEnvironment.application.getSystemService(ACCESSIBILITY_SERVICE);
-    shadowAccessibilityManager = shadowOf(accessibilityManager);
+    shadowAccessibilityManager = shadowOf(getAccessibilityManager());
+  }
+
+  // Emulates Android framework behavior, e.g.,
+  // AccessibilityManager.getInstance(context).isEnabled().
+  private AccessibilityManager getAccessibilityManager() throws Exception {
+    return ReflectionHelpers.callStaticMethod(AccessibilityManager.class, "getInstance",
+            ReflectionHelpers.ClassParameter.from(Context.class, RuntimeEnvironment.application));
   }
 
   @Test
-  public void shouldReturnTrueWhenEnabled() {
+  public void shouldReturnTrueWhenEnabled() throws Exception {
     shadowAccessibilityManager.setEnabled(true);
-    assertThat(accessibilityManager.isEnabled()).isTrue();
+    assertThat(getAccessibilityManager().isEnabled()).isTrue();
   }
 
   @Test
-  public void shouldReturnTrueForTouchExplorationWhenEnabled() {
+  public void shouldReturnTrueForTouchExplorationWhenEnabled() throws Exception {
     shadowAccessibilityManager.setTouchExplorationEnabled(true);
-    assertThat(accessibilityManager.isTouchExplorationEnabled()).isTrue();
+    assertThat(getAccessibilityManager().isTouchExplorationEnabled()).isTrue();
   }
 
   @Test
-  public void shouldReturnExpectedEnabledServiceList() {
+  public void shouldReturnExpectedEnabledServiceList() throws Exception {
     List<AccessibilityServiceInfo> expected = new ArrayList<>(Arrays.asList(new AccessibilityServiceInfo()));
     shadowAccessibilityManager.setEnabledAccessibilityServiceList(expected);
-    assertThat(accessibilityManager.getEnabledAccessibilityServiceList(0)).isEqualTo(expected);
+    assertThat(getAccessibilityManager().getEnabledAccessibilityServiceList(0)).isEqualTo(expected);
   }
 
   @Test
-  public void shouldReturnExpectedInstalledServiceList() {
+  public void shouldReturnExpectedInstalledServiceList() throws Exception {
     List<AccessibilityServiceInfo> expected = new ArrayList<>(Arrays.asList(new AccessibilityServiceInfo()));
     shadowAccessibilityManager.setInstalledAccessibilityServiceList(expected);
-    assertThat(accessibilityManager.getInstalledAccessibilityServiceList()).isEqualTo(expected);
+    assertThat(getAccessibilityManager().getInstalledAccessibilityServiceList()).isEqualTo(expected);
   }
 
   @Test
-  public void shouldReturnExpectedAccessibilityServiceList() {
+  public void shouldReturnExpectedAccessibilityServiceList() throws Exception {
     List<ServiceInfo> expected = new ArrayList<>(Arrays.asList(new ServiceInfo()));
     shadowAccessibilityManager.setAccessibilityServiceList(expected);
-    assertThat(accessibilityManager.getAccessibilityServiceList()).isEqualTo(expected);
+    assertThat(getAccessibilityManager().getAccessibilityServiceList()).isEqualTo(expected);
   }
```

Comment:
Hmm... You mean ShadowAccessibilityManager, right?
I'm not sure how to test "between tests" behavior.

private static ShadowAccessibilityManager sShadow1;
private static ShadowAccessibilityManager sShadow2;

@Test
public void testA() {
  sShadow1 = shadowOf(accessibilityManager);
  if (sShadow2 != null) assertThat(sShadow1).isNotEqualTo(sShadow2);
}

@Test
public void testB() {
  sShadow2 = shadowOf(accessibilityManager);
  if (sShadow1 != null) assertThat(sShadow1).isNotEqualTo(sShadow2);
}

Probably not this. Or we could simply do something like,

@Test
public void testC() {
  assertFalse(shadowOf(accessibilityManager).isEnabled());
}

But is there any way to guarantee the test running order? Otherwise, I'm afraid that such a test can be flaky when the mandate is broken.

Please let me know how better we can test it. Thanks.